### PR TITLE
Load SAEBench Matryoshka SAEs

### PR DESCRIPTION
**Pull request by pleask**
# Description

The Matryoshka SAEs trained for SAEBench aren't supported by the dictionary learning loader. This commit implements a loader and adds some of the SAEs to the pretrained SAE config.

Also fixes a broken test.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [X] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)

# Checklist:

- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing tests pass locally with my changes
- [X] I have not rewritten tests relating to key interfaces which would affect backward compatibility

### You have tested formatting, typing and tests

- [X] I have run `make check-ci` to check format and linting. (you can run `make format` to format code if needed.)

An automated test is included, and the following script was used to test loading the individual SAEs and check their fraction of variance explained is within range.

```
import torch
from transformer_lens import HookedTransformer
from sae_lens.evals import get_sparsity_and_variance_metrics
from sae_lens.training.activation_scaler import ActivationScaler
from sae_lens.training.activations_store import ActivationsStore
from sae_lens import SAE
from itertools import product

device = "cuda" if torch.cuda.is_available() else "cpu"

sae_ids = [f"blocks.12.hook_resid_post__trainer_{i}" for i in range(6)]
releases = [
    "saebench_gemma-2-2b_width-2pow12_date-0108",
    "saebench_gemma-2-2b_width-2pow14_date-0107",
    "saebench_gemma-2-2b_width-2pow16_date-0107",
]

model = HookedTransformer.from_pretrained("gemma-2-2b", device=device)

for release, sae_id in product(releases, sae_ids):
    sae = SAE.from_pretrained(
        release = release,
        sae_id = sae_id,
        device = device,
    )

    activation_store = ActivationsStore.from_sae(
        model=model,
        sae=sae,
        dataset="NeelNanda/pile-10k",
        dataset_trust_remote_code=False,
        streaming=False,
        store_batch_size_prompts=8,
        device=device,
    )

    with torch.no_grad():
        variance_metrics, _ = get_sparsity_and_variance_metrics(
            sae=sae,
            model=model,
            activation_store=activation_store,
            activation_scaler=ActivationScaler(),
            n_batches=32,
            compute_l2_norms=False,
            compute_sparsity_metrics=True,
            compute_variance_metrics=True,
            compute_featurewise_density_statistics=False,
            eval_batch_size_prompts=32,
            model_kwargs={},
        )

    explained_variance = variance_metrics["explained_variance"]
    sparsity = variance_metrics["l0"]
    print(f"{release:<45} {sae_id:<40} {sparsity:<10} {explained_variance:<15}")
```

### Performance Check.

If you have implemented a training change, please indicate precisely how performance changes with respect to the following metrics:
- [ ] L0
- [ ] CE Loss
- [ ] MSE Loss
- [ ] Feature Dashboard Interpretability

Please links to wandb dashboards with a control and test group.

### CODE REVIEW

**Comment by chanind** [comment]
Thanks for catching this and submitting this fix! I just made some minor changes:

- The matryoshka logic is now handled by the dictionary_learning loader rather than having a separate function for matryoshka SAEs
- We can't load real SAE weights in tests because github actions runs out of disk space and CI crashes 😢. So I mocked the weight loading instead.

I'm going to test this out in a notebook to make sure everything seems correct then merge.

**Comment by chanind** [comment]
It looks like CI is still running out of disk space. That isn't due to this PR, but means that we can't merge until it gets fixed.  I'll merge once I can resolve that issue.
